fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordStoreState.java

```diff
@@ -100,15 +100,25 @@ public void endRead() {
     }
 
     /**
-     * Retrieve the mapping from index names to {@link IndexState} that is
+     * Retrieve the index meta-data of the given index from the underlying {@code RecordStoreState}.
+     * @param indexName the index name
+     * @return the index meta-data
+     */
+    @Nullable
+    public IndexMetaData getIndexMetaData(String indexName) {
+        return Collections.unmodifiableMap(indexMetaDataMap.get()).getOrDefault(indexName, null);
```

Comment:
Why are you going through the expense of creating another `Map` just to extract a value from the original one and throw it away again?